ui-ngx/src/app/modules/home/components/widget/lib/gateway/connectors-configuration/modbus/modbus-basic-config/modbus-basic-config.component.ts

```diff
@@ -108,9 +109,15 @@ export class ModbusBasicConfigComponent implements ControlValueAccessor, Validat
     this.basicFormGroup.setValue(editedBase, {emitEvent: false});
   }
 
-  validate(): ValidationErrors | null {
-    return this.basicFormGroup.valid ? null : {
-      basicFormGroup: {valid: false}
-    };
+  validate(basicConfigControl: UntypedFormControl): ValidationErrors | null {
+    const masterHasSlaves = !!basicConfigControl.value.master?.slaves?.length;
+    const slaveEnabled = basicConfigControl.value.slave?.sendDataToThingsBoard;
+    const slaveIsValid = this.basicFormGroup.get('slave').valid;
```

Comment:
Try using only `basicConfigControl`

Reply:
I think it's not possible to check validity of slave without this.basicFormGroup